cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java

```diff
@@ -60,18 +92,69 @@
         }
       }
     }
+    refreshTopicMinISRCache();
     if (!topicsWithBadReplicationFactor.isEmpty()) {
-      return Collections.singleton(createTopicReplicationFactorAnomaly(topicsWithBadReplicationFactor,
+      maybeRetrieveAndCacheTopicMinISR(topicsWithBadReplicationFactor);
+      Map<Boolean, Set<String>> topicsByFixability = new HashMap<>(2);
+      for (String topic : topicsWithBadReplicationFactor) {
+        if (_cachedTopicMinISR.containsKey(topic)) {
+          short topicMinISR = _cachedTopicMinISR.get(topic).minISR();
+          if (_targetReplicationFactor < topicMinISR + _topicReplicationFactorMargin) {
+            topicsByFixability.putIfAbsent(false, new HashSet<>());
+            topicsByFixability.get(false).add(topic);
+          } else {
+            topicsByFixability.putIfAbsent(true, new HashSet<>());
+            topicsByFixability.get(true).add(topic);
+          }
+        }
+      }
+      return Collections.singleton(createTopicReplicationFactorAnomaly(topicsByFixability,
                                                                        _targetReplicationFactor));
     }
     return Collections.emptySet();
   }
 
-  private TopicAnomaly createTopicReplicationFactorAnomaly(Set<String> topicsWithBadReplicationFactor,
-                                                           int targetReplicationFactor) {
+  /**
+   * Retrieve topic minISR config information if it is not cached locally.
+   * @param topicsToCheck Set of topics to check.
+   */
+  private void maybeRetrieveAndCacheTopicMinISR(Set<String> topicsToCheck) {
+    List<ConfigResource> topicResourcesToCheck = new ArrayList<>(topicsToCheck.size());
+    topicsToCheck.stream().filter(t -> !_cachedTopicMinISR.containsKey(t))
+                          .forEach(t -> topicResourcesToCheck.add(new ConfigResource(ConfigResource.Type.TOPIC, t)));
+    for (Map.Entry<ConfigResource, KafkaFuture<Config>> entry : _adminClient.describeConfigs(topicResourcesToCheck).values().entrySet()) {
```

Comment:
What if `topicResourcesToCheck` is empty -- do we need a call to adminClient in this case?